addons/sale/models/account_move.py

```diff
@@ -74,11 +74,17 @@ def _reverse_moves(self, default_values_list=None, cancel=False):
     def action_post(self):
         #inherit of the function from account.move to validate a new tax and the priceunit of a downpayment
         res = super(AccountMove, self).action_post()
-        line_ids = self.mapped('line_ids').filtered(lambda line: any(line.sale_line_ids.mapped('is_downpayment')))
-        for line in line_ids:
+        downpayment_lines = self.line_ids.sale_line_ids.filtered('is_downpayment')
+        other_so_lines = downpayment_lines.order_id.order_line - downpayment_lines
+        real_invoices = other_so_lines.invoice_lines.move_id
+        for dpl in downpayment_lines:
             try:
-                line.sale_line_ids.tax_id = line.tax_ids
-                line.sale_line_ids.price_unit = line.price_unit
+                dpl.price_unit = sum(
+                    l.price_unit if l.move_id.move_type == 'out_invoice' else -l.price_unit
+                    for l in dpl.invoice_lines
+                    if l.move_id.state == 'posted' and l.move_id not in real_invoices  # don't recompute with the final invoice
+                )
+                dpl.tax_id = dpl.invoice_lines.tax_ids
```

Comment:
Does it make sense to aggregate the taxes coming from multiple invoice lines?
Not sure what the tax of a downpayment is supposed to be

Reply:
**In that version**, the system seems to only handle one invoice per downpayment line on the sale order.
If I add more downpayment line manually in the invoice (choosing the downpayment product), it doesn't change anything and thus adding another tax on that line hasn't any effect.

I also tried the following: creating several downpayment lines on the sale order and give different taxes on the invoice.
The taxes are gather by downpayment line which are per invoice.
--> The tax don't stack together.